tests/CMakeLists.txt

```diff
@@ -16,18 +16,56 @@ if(WIN32)
     endif()
 endif()
 
-set(TESTS "minimal;lifting;pendulum;beam;time_schemes;io;rods;bodies_and_rods;wavekin;quasi_static_chain;wavekin_4;tripod;seafloor")
+set(TESTS 
+    minimal
+    lifting
+    pendulum
+    beam
+    time_schemes
+    io
+    rods
+    bodies_and_rods
+    quasi_static_chain
+    wavekin_4
+    tripod
+    seafloor
+)
+
+function(make_executable test_name)
+    add_executable(${test_name} ${test_name}.cpp)
+    target_link_libraries(${test_name} moordyn)
+    add_dependencies(${test_name} Test_MooringConfigFiles)
+endfunction()
+
+function(make_cpp_executable test_name)
+    make_executable(${test_name})
+    target_include_directories(${test_name} PUBLIC ${CMAKE_BINARY_DIR}/)
+endfunction()
+
 if(USE_VTK)
     set(TESTS "${TESTS};vtk")
+
+    make_cpp_executable(wave_surface)
```

Comment:
Replace this by `set(CPP_TESTS "${CPP_TESTS};wave_surface")`. That way the test will be automagically compiled and added to the tests batch (right now it is ignored)

Reply:
In this case, this was an intentional decision. `wave_surface` doesn't do a lot of useful correctness testing, but does serve as a reference for saving wave elevation data to a vtk format. As well as being a way of testing line interaction with the wave surface as part of a development workflow. For this reason, and the fact that it does a lot of IO, I figured it would be best as an executable that gets created, but not as part of the normal ctest suite. If there is a different way you want to accomplish that sort of thing, I'm certainly open to suggestions.